cpp/src/arrow/flight/sql/odbc/flight_sql/CMakeLists.txt

```
@@ -0,0 +1,209 @@
+            accessors/string_array_accessor.cc
+            accessors/string_array_accessor.h
+            accessors/time_array_accessor.cc
+            accessors/time_array_accessor.h
+            accessors/timestamp_array_accessor.cc
+            accessors/timestamp_array_accessor.h
+            address_info.cc
+            address_info.h
+            flight_sql_auth_method.cc
+            flight_sql_auth_method.h
+            flight_sql_connection.cc
+            flight_sql_connection.h
+            flight_sql_driver.cc
+            flight_sql_get_tables_reader.cc
+            flight_sql_get_tables_reader.h
+            flight_sql_get_type_info_reader.cc
+            flight_sql_get_type_info_reader.h
+            flight_sql_result_set.cc
+            flight_sql_result_set.h
+            flight_sql_result_set_accessors.cc
+            flight_sql_result_set_accessors.h
+            flight_sql_result_set_column.cc
+            flight_sql_result_set_column.h
+            flight_sql_result_set_metadata.cc
+            flight_sql_result_set_metadata.h
+            flight_sql_ssl_config.cc
+            flight_sql_ssl_config.h
+            flight_sql_statement.cc
+            flight_sql_statement.h
+            flight_sql_statement_get_columns.cc
+            flight_sql_statement_get_columns.h
+            flight_sql_statement_get_tables.cc
+            flight_sql_statement_get_tables.h
+            flight_sql_statement_get_type_info.cc
+            flight_sql_statement_get_type_info.h
+            flight_sql_stream_chunk_buffer.cc
+            flight_sql_stream_chunk_buffer.h
+            get_info_cache.cc
+            get_info_cache.h
+            json_converter.cc
+            json_converter.h
+            record_batch_transformer.cc
+            record_batch_transformer.h
+            scalar_function_reporter.cc
+            scalar_function_reporter.h
+            system_trust_store.cc
+            system_trust_store.h
+            utils.cc)
+target_include_directories(arrow_odbc_spi_impl PUBLIC ${CMAKE_CURRENT_LIST_DIR})
+if(WIN32)
+  target_sources(arrow_odbc_spi_impl
+                 include/flight_sql/config/configuration.h
+                 include/flight_sql/config/connection_string_parser.h
+                 include/flight_sql/ui/add_property_window.h
+                 include/flight_sql/ui/custom_window.h
+                 include/flight_sql/ui/dsn_configuration_window.h
+                 include/flight_sql/ui/window.h
+                 config/configuration.cc
+                 config/connection_string_parser.cc
+                 ui/custom_window.cc
+                 ui/window.cc
+                 ui/dsn_configuration_window.cc
+                 ui/add_property_window.cc
+                 system_dsn.cc)
+endif()
+
+find_package(ArrowFlightSql)
+
+target_link_libraries(arrow_odbc_spi_impl PUBLIC odbcabstraction
+                                                 ArrowFlightSql::arrow_flight_sql_static)
```

```suggestion
target_link_libraries(arrow_odbc_spi_impl PUBLIC odbcabstraction
                                                 arrow_flight_sql_static)
```
`ArrowFlightSql::` should be removed here. According to [arrow docs](https://arrow.apache.org/docs/developers/cpp/windows.html#statically-linking-to-arrow-on-windows), we can just specify `arrow_flight_sql_static` in `target_link_libraries`. 🙂